### PR TITLE
Add support for OmniSciDB 5.2

### PR DESCRIPTION
**Pull request by jclay**
### Overview

This makes the following changes to reflect the updates made on the `omniscidb` side:

- Updated dependencies to reflect those used in omniscidb 5.2. See note below regarding cudf.
- Updated thrift bindings to reflect the namespace changes from mapd to omnisci.
- Minor updates to documentation to reflect the above

### Current Status

100% of CPU tests are passing. GPU support works for scalar valued projections. ~~but support for queries returning a `dict` column are not supported. Currently `pymapd/_parsers.py:: _parse_tdf_gpu` will throw when the call to `GPUArrowReader` is made. It reports it was not able to find the dictionaries in the stream. Some pre-processing steps need to be performed as we are doing in the `omniscidb` C++ codebase, but I have yet to discover a way to process the dictionary memos with the version of `pyarrow` cudf ships (0.15).~~ Support for text encoded columns has been added with a workaround.

### CODE REVIEW

File: docs/source/contributing.rst [hunk 1]
```diff
@@ -118,19 +131,25 @@ Updating Apache Thrift Bindings
...
+   cd ./omniscidb
+
+   # Use Thrift to generate the Python bindings
+   thrift -gen py omnisci.thrift
    thrift -gen py completion_hints.thrift
    thrift -gen py common.thrift
-   thrift -gen py serialized_result_set.thrift
-   thrift -gen py extension_functions.thrift
+   thrift -gen py QueryEngine/serialized_result_set.thrift
+   thrift -gen py QueryEngine/extension_functions.thrift
```
Comment (andrewseidl) [inline docs/source/contributing.rst:148]: Not blocking, but this whole block could be replaced with `thrift -r -gen py omnisci.thrift`

File: environment.yml [hunk 2]
```diff
@@ -3,15 +3,19 @@ channels:
...
 - coverage
 - flake8
-- "pytest>=3.6,<4.0"
 - pytest-cov
 - pytest-mock
-- rbc
+- shapely
+- sphinx 
+- sphinx_rtd_theme
+- 'rbc==0.2.2'
```
Comment (andrewseidl) [inline environment.yml:21]: Missing newline at eof

File: environment_gpu.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+- numpy
+- numpydoc
+- coverage
+- flake8
+- pytest-cov
+- pytest-mock
+- shapely
+- sphinx 
+- sphinx_rtd_theme
+- 'rbc==0.2.2'
```
Comment (andrewseidl) [inline environment_gpu.yml:25]: Missing newline at eof

**Comment by andrewseidl** [comment]
Also the flake8 warnings look valid https://jenkins-os.mapd.com/blue/organizations/jenkins/pymapd-tests/detail/pymapd-tests/138/pipeline (`black` should be able to fix them for you)

File: docs/source/contributing.rst (hunk 1 above)
Comment (jclay) [reply docs/source/contributing.rst:148]: Oh, great suggestion!

File: pymapd/_parsers.py [hunk 4]
```diff
@@ -151,22 +151,49 @@ def _parse_tdf_gpu(tdf):
...
+        for f in schema:
+            if (pa.types.is_dictionary(f.type)):
+                msg = dict_batch_reader.read_next_batch()
+                dict_memo[f.name] = msg.column(0)
+                updated_fields.append(pa.field(f.name, f.type.index_type))
+            else:
+                updated_fields.append(pa.field(f.name, f.type))
+
+        schema = pa.schema(updated_fields)
+    except pa.ArrowInvalid:
```
Comment (jclay) [inline pymapd/_parsers.py:176]: Need to come up with a better way to catch this.

**Comment by andrewseidl** [comment]
It's strange that Jenkins is happy with pip, the tests are failing for me locally.

I believe we should remove `cudf` from the pip installs since they have stopped maintaining builds there (for good reason), so we end up pulling in cudf 0.6.1.

Otherwise looking good so far in my testing.

**Pull request merged**